slack-api-client/src/main/java/com/slack/api/methods/SlackApiRequest.java

```diff
@@ -17,6 +17,25 @@
  */
 public interface SlackApiRequest {
 
+    /**
+     * Returns a token in this request object.
+     * If the API endpoint does not require a token (e.g., api.test), this method can return null.
+     *
+     * @return token string value
+     */
     String getToken();
 
+    /**
+     * Updates the token in this request object.
+     * <p>
+     * The default implementation throws {@link UnsupportedOperationException}.
+     * All the built-in implementing classes overrides this method in a proper way.
+     *
+     * @throws UnsupportedOperationException is always thrown if this method is not overridden
+     */
+    default void setToken(String token) {
+        throw new UnsupportedOperationException(
+                "SlackApiRequest#setToken(String) is not overridden in this class.");
```

Comment:
I suggest replacing `#` with `.`.  Users will be able to click on the stack frame to jump to this method.  I am not sure what `#` would do besides make it harder for a person to read.

Reply:
Thanks for the suggestion but I am not going to change this. Using # is a common practice when meeting an instance method. If I say `SlackApiRequest.setToken`, it indicates the method may be a static/class method.